# DataFrame Analysis Exercise

**Import the `bestsellers.csv` dataset and use it to answer the following questions:**

* Find the lowest User Rating in the DF
* Find the highest Price in the DF
* What is the average User Rating?
* What the average User Rating of the first 5 rows?
* What User Review score appeared the most?
* What is the total (sum) of all the values in the Reviews column?
* How many different authors are featured in the dataset?
* Which author wrote the most number of books on the list?  How many did they write?

In [1]:
import pandas as pd


In [2]:
df=pd.read_csv("./data/bestsellers.csv")

In [6]:
df.describe(include=["O","int64"])

Name       Author  \
count                                                 550          550   
unique                                                351          248   
top     Publication Manual of the American Psychologic...  Jeff Kinney   
freq                                                   10           12   
mean                                                  NaN          NaN   
std                                                   NaN          NaN   
min                                                   NaN          NaN   
25%                                                   NaN          NaN   
50%                                                   NaN          NaN   
75%                                                   NaN          NaN   
max                                                   NaN          NaN   

             Reviews       Price         Year        Genre  
count     550.000000  550.000000   550.000000          550  
unique           NaN         NaN          NaN            2  
top              NaN         NaN          NaN  Non Fiction  
freq             NaN         NaN          NaN          310  
mean    11953.281818   13.100000  2014.000000          NaN  
std     11731.132017   10.842262     3.165156          NaN  
min        37.000000    0.000000  2009.000000          NaN  
25%      4058.000000    7.000000  2011.000000          NaN  
50%      8580.000000   11.000000  2014.000000          NaN  
75%     17253.250000   16.000000  2017.000000          NaN  
max     87841.000000  105.000000  2019.000000          NaN

1. lowest user Rating movie ** 3.300000**
2. higest price **105**
3. avg user rating **4.618364**
4. the average User Rating of the first 5 rows **4.7**
5. total review **550**
6. most no of book written by **Jeff Kinney**
7. uniquea auther **248**

In [9]:
df["User Rating"].head().describe()

count    5.000000
mean     4.700000
std      0.070711
min      4.600000
25%      4.700000
50%      4.700000
75%      4.700000
max      4.800000
Name: User Rating, dtype: float64

In [11]:
df.sum()

Name           10-Day Green Smoothie Cleanse11/22/63: A Novel...
Author         JJ SmithStephen KingJordan B. PetersonGeorge O...
User Rating                                               2540.1
Reviews                                                  6574305
Price                                                       7205
Year                                                     1107700
Genre          Non FictionFictionNon FictionFictionNon Fictio...
dtype: object